### PR TITLE
[WIP] debridge frontend

### PR DESCRIPTION
**Pull request by enu-kuro**

### CODE REVIEW

File: pwa/src/views/HomeView.vue [hunk 1]
```diff
@@ -975,89 +961,33 @@ const bridgeNFT = async () => {
...
+      })?.value;
+      console.log(chainIdFrom);
+      await ethereum.request({
+        method: "wallet_switchEthereumChain",
+        params: [{ chainId: BigNumber.from(chainIdFrom).toHexString() }],
+      });
+      // TODO:
+      const nftContractAddress = "0x03e055692e77e56aBf7f5570D9c64C194BA15616";
+      const tokenId = 5001;
+      const receipt = await bridge(
```
Comment (enu-kuro) [inline pwa/src/views/HomeView.vue:981]: how to get nft contract address and token id?

File: pwa/src/views/HomeView.vue [hunk 2]
```diff
@@ -86,11 +86,7 @@
               </span>
             </div>
             <div class="button-container">
-              <button
-                class="bridge-button-left"
-                @click="bridgeNFT()"
-                :disabled="!approvedBridge"
```
Comment (enu-kuro) [inline pwa/src/views/HomeView.vue:92]: temporarily removed. 
And you don't need approve button.
approve and send(bridge) are included in the same function.

File: pwa/src/services/debridge.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+const isSupported = (chainId) => {
+  return (
+    Object.keys(rpcNodes).filter(
+      (availableChainId) => availableChainId.toString() === chainId.toString()
+    ).length > 0
+  );
+};
+
+// TODO: how should manage txHash?
```
Comment (enu-kuro) [inline pwa/src/services/debridge.js:21]: You need txHash when claiming.
Managing txHash properly seems very tiring.
Storing in LocalStorage is a convenient way.

**Comment by enu-kuro** [comment]
Currently support only Polygon and Arbirtum mainnet.

**Comment by enu-kuro** [comment]
I don't know how I should handle errors.

File: pwa/src/views/HomeView.vue (hunk 2 above)
Comment (Webmoss) [reply pwa/src/views/HomeView.vue:92]: This just locks the NFT once you have selected it, so it disables the bridge button until the user approves, almost like a safety measure on UI, once it's locked then the Bridge button will work

File: pwa/src/views/HomeView.vue (hunk 1 above)
Comment (Webmoss) [reply pwa/src/views/HomeView.vue:980]: see /* Bridge NFT Details */
const tokenIdBridge = ref("");
const contractAddressBridge = ref("");
const nameBridge = ref("");
const imageUrlBridge = ref(null);

Comment (Webmoss) [reply pwa/src/views/HomeView.vue:980]: tokenIdBridge.value will give you the NFT token id
contractAddressBridge.value the NFT's contract address

inside bridgeNFT function you can use those, I've got it setup like a mint function so gets signer and will run a txn, but let me know and we can sort that bit out

File: pwa/src/services/debridge.js (hunk 3 above)
Comment (Webmoss) [reply pwa/src/services/debridge.js:21]: we can do this via bridgeNFT() or in the service.

File: pwa/src/views/HomeView.vue (hunk 2 above)
Comment (enu-kuro) [reply pwa/src/views/HomeView.vue:92]: I made a mistake...
This comment should have been for the code below.

we don't need "approve" button.
When you click "bridge" button, if your NFT haven't approved, metamask's approval popup will show up, and if already approved, just skip it.

```
            <div class="nft-bridge-modal-approve">
              <button
                :class="!approvedBridge ? 'approve-button' : 'approved-button'"
                @click="ConfirmApprovedBridge(true)"
              >
                {{ !approvedBridge ? "Approve" : "Locked" }}
              </button>
              <button class="cancel-button" @click="CancelBridge()">
                Cancel
              </button>
            </div>
```

File: pwa/src/services/debridge.js [hunk 4]
```diff
@@ -22,10 +23,15 @@ const isSupported = (chainId) => {
 // not the best solution...
 const TX_HASH_LOCAL_STORAGE_KEY = "debridge_tx_hash";
 const storeTxHash = (txHash) => {
```
Comment (enu-kuro) [inline pwa/src/services/debridge.js:25]: The reason why I use LocalStorage is that users can reload page and browser states vanish.
Confirmation takes a long time. Probably on polygon it's more than 10 minutes?
https://docs.debridge.finance/build-with-debridge/getting-started#checking-the-status-of-the-submission

But maybe, just for Demo, we don't need to consider reloading.

**Pull request merged**